packages/discovery-provider/src/tasks/entity_manager/entities/playlist.py

```diff
@@ -190,6 +198,74 @@ def update_playlist_tracks(params: ManageEntityParameters, playlist_record: Play
     )
 
 
+def update_album_price_history(
+    session: Session,
+    playlist_record: Playlist,
+    playlist_metadata: dict,
+    blocknumber: int,
+    timestamp: datetime,
+):
+    """Adds an entry in the track price history table to record the price change of a track or change of splits if necessary."""
+    new_record = None
+    is_stream_gated = playlist_metadata.get(
```

Comment:
Just curious if this is supposed to intersect with the download-gating logic? I'm not up to speed on the feature spec here, just wondering if we're supposed to support purchasable downloads and if this code feels set up to add that extension without a lot of refactoring.